### PR TITLE
dotCMS/core#26912 [UI] AI actions menu selection is not working as should.

### LINKED ISSUE #26912: [UI] AI actions menu selection is not working as should.
### Parent Issue
#26495

### User Story

The actions menu selected option is not working as should. 

image

### Acceptance Criteria
- Should mark just one active option. 

### dotCMS Version
master

### Proposed Objective
Integrations

### Proposed Priority
Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
_to_define_

### Assumptions & Initiation Needs
- It is assumed that the structure of the plugin's endpoint has changed significantly.

### Quality Assurance Notes & Workarounds
- Thorough testing to validate compatibility with the new endpoint structure.

### Sub-Tasks & Estimates
_to_define_

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* double selection on the menu not happening. 
* show a loading indicator while waiting a response from the service when re-generating. 

### Screenshots

https://github.com/dotCMS/core/assets/31667212/12ffae01-aa52-44ae-b608-a678d81c7c83

### CODE REVIEW

File: core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.scss [hunk 1]
```diff
@@ -278,6 +278,17 @@
                     }
                 }
             }
+
+            .ai-loading {
+                display: flex;
+                justify-content: center;
+                align-items: center;
+                min-width: 100%;
+                padding: 0.5rem;
```
Comment (manuelrojas) [inline core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.scss:287]: what about using the variables for spacing?

Comment (hmoreras) [reply core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.scss]: done

**Review by manuelrojas — APPROVED** [review]

File: core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts [hunk 2]
```diff
@@ -127,15 +127,13 @@ export class AIContentPromptView {
...
 
-        this.componentStore.status$
-            .pipe(
-                skip(1),
-                takeUntil(this.destroy$),
-                filter((status) => status === 'exit')
-            )
-            .subscribe(() => {
+        this.componentStore.status$.pipe(skip(1), takeUntil(this.destroy$)).subscribe((status) => {
+            if (status === 'exit') {
```
Comment (oidacra) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts:131]: What do you think about instead use a string, use the const enum about componentStates?

File: core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts [hunk 3]
```diff
@@ -127,15 +127,13 @@ export class AIContentPromptView {
...
                 this.tippy?.hide();
-            });
+            } else if (status === 'loading') {
```
Comment (oidacra) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts:133]: same here.

File: core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts (hunk 2 above)
Comment (hmoreras) [reply core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts:135]: agree ! done

File: core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts (hunk 3 above)
Comment (hmoreras) [reply core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts:137]: agree ! done

**Review by oidacra — APPROVED** [review]

**Pull request merged**